### PR TITLE
2.0.0 release blog post

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

File: _posts/2020-10-15-2.0.0-release.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+## Arrow Flight RPC notes
+
+## C++ notes
+
+## C# notes
+
+## Go notes
+
+## Java notes
+
```
Comment (liyafan82) [inline _posts/2020-10-15-2.0.0-release.md:62]: @nealrichardson Please check if the following notes are reasonable for Java. Thanks for your effort.

The Java package has supported a number of new features. 
Users can validate vectors in a wider range of aspects, if they are willing to take more time. 
In dictionary encoding, dictionary indices can be expressed as unsinged integers.
A framework for data compression has been setup for IPC.  

The calculation for vector capacity has been simplified, so users should experience notable performance
improvements for various 'setSafe' methods. 

Bugs for JDBC adapters, sort algorithms, and ComplexCopier have been resolved to make them more usable.

File: _posts/2020-10-15-2.0.0-release.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+you to the [complete changelog][3].
+
+## Community
+
+<!-- Acknowledge and link to any new committers and PMC members since the last release. See previous release announcements for examples. -->
+
+## Columnar Format Notes
+
+## Arrow Flight RPC notes
+
```
Comment (lidavidm) [inline _posts/2020-10-15-2.0.0-release.md:54]:
```suggestion
For Arrow Flight, 2.0.0 mostly brings bugfixes. In Java, some memory leaks in `FlightStream` and `DoPut` have been addressed. In C++ and Python, a deadlock has been fixed in an edge case.
```

File: _posts/2020-10-15-2.0.0-release.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+you to the [complete changelog][3].
+
+## Community
+
+<!-- Acknowledge and link to any new committers and PMC members since the last release. See previous release announcements for examples. -->
+
+## Columnar Format Notes
+
+## Arrow Flight RPC notes
+For Arrow Flight, 2.0.0 mostly brings bugfixes. In Java, some memory leaks in `FlightStream` and `DoPut` have been addressed. In C++ and Python, a deadlock has been fixed in an edge case.
```
Comment (lidavidm) [inline _posts/2020-10-15-2.0.0-release.md:54]:
```suggestion
For Arrow Flight, 2.0.0 mostly brings bugfixes. In Java, some memory leaks in `FlightStream` and `DoPut` have been addressed. In C++ and Python, a deadlock has been fixed in an edge case. Additionally, when supported by gRPC, TLS verification can be disabled.
```

**Comment by jorisvandenbossche** [comment]
I did a first pass on adding a few python notes

File: _posts/2020-10-15-2.0.0-release.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+The `pyarrow.compute` module now has a complete coverage of the available C++
+compute kernels in the python API. Several new kernels have been added (... TODO
+list a few ...).
+
+The `pyarrow.dataset` module was further improved. In addition to reading,
+it is now also possible to write partitioned datasets (with `write_dataset()`).
+
+The Arrow <-> Python conversion code was refactored, fixing several bugs
+and corner cases.
+
```
Comment (BryanCutler) [inline _posts/2020-10-15-2.0.0-release.md:96]: Conversion of an array of `pyarrow.MapType` to Pandas has been added.

File: _posts/2020-10-15-2.0.0-release.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+## R notes
+
+
+For more on what’s in the 2.0.0 R package, see the [R changelog][4].
+
+## Ruby and C GLib notes
+
+### Ruby
+
```
Comment (mrkn) [inline _posts/2020-10-15-2.0.0-release.md:106]:
```suggestion

In Ruby binding, `Arrow::Table#save` uses the number of rows as the `chunk_size` parameter by default when the table is saved in a Parquet file.

```

File: _posts/2020-10-15-2.0.0-release.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+### C GLib
+
```
Comment (mrkn) [inline _posts/2020-10-15-2.0.0-release.md:108]:
```suggestion

The GLib binding newly supports `GArrowStringDictionaryArrayBuilder` and `GArrowBinaryDictionaryArrayBuilder`.

Moreover the GLib binding supports new accessors of `GArrowListArray` and `GArrowLargeListArray`.  They are `get_values`, `get_value_offset`, `get_value_length`, and `get_value_offsets`.

```

**Review by mrkn — CHANGES_REQUESTED** [review]
I added two change suggestions for Ruby and GLib notes.

File: _posts/2020-10-15-2.0.0-release.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Many other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## Community
+
+<!-- Acknowledge and link to any new committers and PMC members since the last release. See previous release announcements for examples. -->
+
+## Columnar Format Notes
```
Comment (emkornfield) [inline _posts/2020-10-15-2.0.0-release.md:51]: I think technically Decimal256 went into the format but not the code base.  Not sure if we want to bump the minor version now or when library support is present next release.

File: _posts/2020-10-15-2.0.0-release.md [hunk 8]
```diff
@@ -80,6 +80,15 @@ Bugs for JDBC adapters, sort algorithms, and ComplexCopier have been resolved to
...
 ## Python notes
 
+Parquet reading now supports round trip of arbitrarily nested data. Several bug fixes
+for writing nested data and FixedSizeList.  If writing data with these type we recommend validating 
+old data (there is potential some data loss) and upgrade to 2.0.
+
+Extension types now round trip through Parquet.
+
+Conversion of Timezone aware datetimes to and/from pyarrow arrays including pandas now round-trip preserving timezone.  To use the old behavior (e.g. for spark) set the environment variable 
+PYARROW_IGNORE_TIMEZONE
```
Comment (wesm) [inline _posts/2020-10-15-2.0.0-release.md:90]: This has to be set to equal 1 (or some truthy value?) I think

**Comment by wesm** [comment]
Since this is the first release we've made where the Format and Library versions have diverged, we should put a dedicated section calling this out so people don't think that the major version bump means any possibly format-level incompatibilities

File: _posts/2020-10-15-2.0.0-release.md (hunk 8 above)
Comment (emkornfield) [reply _posts/2020-10-15-2.0.0-release.md]: yes.  clarified.

File: _posts/2020-10-15-2.0.0-release.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+## Community
+
+<!-- Acknowledge and link to any new committers and PMC members since the last release. See previous release announcements for examples. -->
+
+## Columnar Format Notes
+
+## Arrow Flight RPC notes
+For Arrow Flight, 2.0.0 mostly brings bugfixes. In Java, some memory leaks in `FlightStream` and `DoPut` have been addressed. In C++ and Python, a deadlock has been fixed in an edge case. Additionally, when supported by gRPC, TLS verification can be disabled.
+## C++ notes
+
```
Comment (nealrichardson) [inline _posts/2020-10-15-2.0.0-release.md:56]: @pitrou @bkietz @emkornfield we need C++ notes. Presumably the nested parquet should be called out here, along with dataset writing, and any other highlights.

File: _posts/2020-10-15-2.0.0-release.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+from [**XX distinct contributors**][2]. See the Install Page to learn how to
+get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Many other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## Community
+
+<!-- Acknowledge and link to any new committers and PMC members since the last release. See previous release announcements for examples. -->
```
Comment (nealrichardson) [inline _posts/2020-10-15-2.0.0-release.md:49]: Need to list Jorge here

File: _posts/2020-10-15-2.0.0-release.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+In Ruby binding, `Arrow::Table#save` uses the number of rows as the `chunk_size` parameter by default when the table is saved in a Parquet file.
+
+### C GLib
+
+The GLib binding newly supports `GArrowStringDictionaryArrayBuilder` and `GArrowBinaryDictionaryArrayBuilder`.
+
+Moreover the GLib binding supports new accessors of `GArrowListArray` and `GArrowLargeListArray`.  They are `get_values`, `get_value_offset`, `get_value_length`, and `get_value_offsets`.
+
+## Rust notes
+
```
Comment (nealrichardson) [inline _posts/2020-10-15-2.0.0-release.md:134]: @andygrove what's the status of your Rust blog post? If it's not ready right yet, maybe make a note that there's a lot of changes and there will be a followup post specifically about Rust

File: _posts/2020-10-15-2.0.0-release.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+* Update all "XX" values with the appropriate numbers (you can get the resolved issues and contributors count from `_release/2.0.0.md`)
+* Fill in the various sections below. Note that the audience is the broader user community, not Arrow developers, so please write clearly using terms they will understand and care about. Delete any sections that don't have any content (as in, there are no changes to announce)
+* Delete this introductory comment
+
+ -->
+
+
+The Apache Arrow team is pleased to announce the 2.0.0 release. This covers
+over XX months of development work and includes [**XX resolved issues**][1]
+from [**XX distinct contributors**][2]. See the Install Page to learn how to
```
Comment (nealrichardson) [inline _posts/2020-10-15-2.0.0-release.md:40]: @kszucs can you fill these numbers in now?

File: _posts/2020-10-15-2.0.0-release.md (hunk 11 above)
Comment (andygrove) [reply _posts/2020-10-15-2.0.0-release.md:134]: @nealrichardson I just created the PR: https://github.com/apache/arrow-site/pull/81

File: _posts/2020-10-15-2.0.0-release.md (hunk 12 above)
Comment (kszucs) [reply _posts/2020-10-15-2.0.0-release.md:40]: Available in the [release note pr](https://github.com/apache/arrow-site/pull/80/files#diff-0f3f18b3362ba5c10197b69c36d78008aa794c049f82b015f89ce282a37dc927R41) and [JIRA release page](https://issues.apache.org/jira/projects/ARROW/versions/12345977).

File: _posts/2020-10-15-2.0.0-release.md [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+In dictionary encoding, dictionary indices can be expressed as unsigned integers.
+A framework for data compression has been setup for IPC.  
+
+The calculation for vector capacity has been simplified, so users should experience notable performance
+improvements for various 'setSafe' methods.
+
+Bugs for JDBC adapters, sort algorithms, and ComplexCopier have been resolved to make them more usable.
+
+## JavaScript notes
+
```
Comment (nealrichardson) [inline _posts/2020-10-15-2.0.0-release.md:74]: Need JS notes: @trxcllnt can you please summarize in a sentence or two?

Comment (trxcllnt) [inline _posts/2020-10-15-2.0.0-release.md:74]:
```suggestion
Upgrades Arrow's build to use TypeScript 3.9, fixing generated `.d.ts` typings.
```

File: _posts/2020-10-15-2.0.0-release.md (hunk 9 above)
Comment (emkornfield) [reply _posts/2020-10-15-2.0.0-release.md:47]: yes, for parquet.  I think you can copy my comments above about parquet from the python section.

File: _posts/2020-10-15-2.0.0-release.md (hunk 7 above)
Comment (nealrichardson) [reply _posts/2020-10-15-2.0.0-release.md:40]: Good question; for now I'll remove this section and we can announce Decimal256 in the next release when there is some support for it.

**Comment by wesm** [comment]
> Since this is the first release we've made where the Format and Library versions have diverged, we should put a dedicated section calling this out so people don't think that the major version bump means any possibly format-level incompatibilities

Should we add something about this? The Format/Library version split may not have sunk in with very many people.

**Comment by wesm** [comment]
I added the note about versioning. I think we're pretty good here

**Comment by wesm** [comment]
Shall we publish?

**Comment by nealrichardson** [comment]
I'll click "squash and merge" once I finish catching up on my other notifications, unless you've merged first.

**Pull request merged**